frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,156 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
+                text = _("Add custom message to sleep screen"),
+                checked_func = function()
+                    return G_reader_settings:isTrue("screensaver_show_message")
+                end,
                 callback = function()
-                    Screensaver:chooseFile(true)
+                    G_reader_settings:toggle("screensaver_show_message")
                 end,
+                separator = true,
             },
             {
-                text = _("Screensaver message"),
-                keep_menu_open = true,
+            text = _("Edit sleep screen message"),
+            enabled_func = function()
+                return G_reader_settings:isTrue("screensaver_show_message")
+            end,
+            keep_menu_open = true,
                 callback = function()
                     Screensaver:setMessage()
                 end,
             },
             {
-                text = _("Covers and images settings"),
+                text = _("Background fill"),
+                help_text = _([[This option will only become available, if you have selected 'Put screen to sleep in current state'
+                    as screensaver and have 'Sleep screen message' on.]]),
+               enabled_func = function()
+                    return G_reader_settings:readSetting("screensaver_type") == "disable" and G_reader_settings:isTrue("screensaver_show_message")
+                end,
                 sub_item_table = {
-                    genMenuItem(_("Black background"), "screensaver_img_background", "black"),
-                    genMenuItem(_("White background"), "screensaver_img_background", "white"),
-                    genMenuItem(_("Leave background as-is"), "screensaver_img_background", "none", nil, true),
-                    -- separator
-                    {
-                        text_func = function()
-                            local percentage = G_reader_settings:readSetting("screensaver_stretch_limit_percentage")
-                            if G_reader_settings:isTrue("screensaver_stretch_images") and percentage then
-                                return T(_("Stretch to fit screen (with limit: %1 %)"), percentage)
-                            end
-                            return _("Stretch to fit screen")
-                        end,
-                        checked_func = function()
-                            return G_reader_settings:isTrue("screensaver_stretch_images")
-                        end,
-                        callback = function(touchmenu_instance)
-                            Screensaver:setStretchLimit(touchmenu_instance)
-                        end,
-                    },
+                    genMenuItem(_("Black fill"), "screensaver_msg_background", "black"),
+                    genMenuItem(_("White fill"), "screensaver_msg_background", "white"),
+                    genMenuItem(_("Current state"), "screensaver_msg_background", "none", nil, true),
```

Comment:
So, for the wallpaper, we currently have "Border fill>" with just colors: "Black", "White", "Current state".
While here for message we have "Background fill>" with "Black fill", "White fill"...
Feels a bit inconsistent.
(I prefer this version, with "fill" - but not really caring.)